src/core/public/chrome/doc_title/doc_title_service.ts

```diff
@@ -0,0 +1,180 @@
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { BehaviorSubject, Observable, ReplaySubject } from 'rxjs';
+import { compact, flattenDeep, isString, isArray } from 'lodash';
+import { takeUntil } from 'rxjs/operators';
+
+interface StartDeps {
+  document: { title: string };
+}
+
+/**
+ * {@link ChromeDocTitle | APIs} for accessing and updating the document title.
+ *
+ * @example
+ * How to change the title of the document
+ * ```ts
+ * chrome.docTitle.change({ parts: ['My application'], excludeBase: true })
+ * ```
+ *
+ * @example
+ * How to reset the title of the document to it's initial value
+ * ```ts
+ * chrome.docTitle.reset()
+ * ```
+ *
+ * @public
+ * */
+export interface ChromeDocTitle {
+  /**
+   * Gets an observable of the current document title.
+   */
+  get$: () => Observable<string>;
+  /**
+   * Changes the current document title.
+   *
+   * @remarks
+   * The apply option is mostly here for legacy compatibility reasons and it's default value
+   * should probably always be used.
+   *
+   * @example
+   * How to change the title of the document
+   * ```ts
+   * chrome.docTitle.change('My application title')
+   * chrome.docTitle.change(['My application', 'My section'])
+   * chrome.docTitle.change({ parts: ['My application'], excludeBase: true })
+   * ```
+   *
+   * @param newTitle The new title to set, either a string, string array or {@link ChromeDocTitleEntry}
+   * @param [apply=true] If false, will not actually apply the new title until #apply() is called.
+   */
+  change: (newTitle: ChromeDocTitleChange, apply?: boolean) => void;
+  /**
+   * Resets the document title to it's initial value.
+   * (meaning the one present in the title meta at application load.)
+   *
+   * @remarks
+   * The apply option is mostly here for legacy compatibility reasons and it's default value
+   * should probably always be used.
+   *
+   * @param [apply=true] If false, will not actually apply the new title until #apply() is called.
+   */
+  reset: (apply?: boolean) => void;
+  /**
+   * Apply a title change or reset that was not applied yet.
+   *
+   * @see {DocTitle#change}
+   * @see {DocTitle#reset}
+   */
+  apply: () => void;
+  /** @internal */
+  __legacy: {
+    setBaseTitle: (baseTitle: string) => void;
+  };
+}
+
+/** @public */
+export interface ChromeDocTitleEntry {
+  parts: string[];
+  excludeBase?: boolean;
+}
+
+/**
+ * Composed type for the {@link ChromeDocTitle#change} possible inputs.
+ *
+ * @public
+ */
+export type ChromeDocTitleChange = string | string[] | ChromeDocTitleEntry;
```

Comment:
This is a only a nice to have, but it felt like this could ease the developer way to call the API. If we prefer something more rigid, we can stick to `ChromeDocTitleEntry` instead

Reply:
Now that I think more about this... this service does just two things, if you give it parts it joins them with ` - ` and it adds the baseTitle. It's so small it's almost annoying :-P but the value it provides is consistency for all of Kibana. If a plugin really needs to break this consistency it feels like they might as well just use `document.title = `.

Reply:
The ingest team is happy for us to drop the `true` flag in this PR so that monitoring also uses the baseTitle.

Reply:
@rudolf ok great. then I guess I will change the type to `export type ChromeDocTitleChange = string | string[]`

Reply:
But yes, the dual logic between displaying the title and the `join(' - ')` is quite annoying :)